src/utils/lbann_library.cpp

```diff
@@ -182,16 +182,16 @@ std::unique_ptr<model> build_model_from_prototext(
 
   // Initialize models differently if needed.
 #ifndef LBANN_DETERMINISTIC
-  if (pb_model->random_init_models_differently()) {
+  if (!pb_trainer->random_init_trainers_identically()) {
     random_seed = random_seed + comm->get_trainer_rank();
```

Comment:
Perhaps beyond the scope of this PR, but we have a [utility function to mix RNG seeds](https://github.com/LLNL/lbann/blob/279b4d4a5f0268519c9ce91f5940a59a355d1336/include/lbann/utils/hash.hpp#L47):
```suggestion
     random_seed = hash_combine(random_seed, comm->get_trainer_rank());
```

Reply:
I will apply this.